src/System.Runtime/tests/System/ArrayTests.cs

```diff
@@ -533,6 +550,29 @@ public static IEnumerable<object[]> Copy_TestData()
         }
 
         [Theory]
+        public static void ForEach()
+        {
+            Array.ForEach<int>(new int[] { }, new Action<int>(i => { throw new InvalidOperationException(); }));
+            // Did not throw; no items
+
+            int counter = 0;
+            Array.ForEach<int>(new int[] { 1, 2, 3 }, new Action<int>(i => counter += i));
+            Assert.Equal(6, counter);
+
+            // Only works on one dimensional arrays
```

Comment:
@stephentoub I'm not sure what you mean. Testing that Foreach doesn't work for multi-dimensional arrays? You can technically do something like: 
```
int[][] array = new int[][] { new int[] { 1, 2}, new int[] { 3, 4 } };
int counter = 0;
Array.ForEach<int[]>(array, new Action<int[]>(arr => counter += arr[0] + arr[1] ));
```
but anything other than that would be syntactically incorrect. Please correct me if what I assumed is not relevant.